### PR TITLE
[cli] generate command to create scaffolding

### PR DESCRIPTION
**Pull request by mikz**
The idea is to replace the need to copy-paste with scaffolding.

```shell
$ bin/apicast generate policy somename -f
source: /Users/mikz/Developer/3scale/apicast/examples/scaffold/policy
destination: /Users/mikz/Developer/3scale/apicast

created: spec
created: spec/policy
created: spec/policy/somename
created: spec/policy/somename/somename_spec.lua
created: t
created: t/apicast-policy-somename.t
created: gateway
created: gateway/src
created: gateway/src/apicast
created: gateway/src/apicast/policy
created: gateway/src/apicast/policy/somename
created: gateway/src/apicast/policy/somename/apicast-policy.json
created: gateway/src/apicast/policy/somename/init.lua
created: gateway/src/apicast/policy/somename/somename.lua
```

### CODE REVIEW

File: examples/scaffold/policy/gateway/src/apicast/policy/{{policy.file}}/{{policy.file}}.lua [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- This is a {{ policy.name }} description.
+
+local policy = require('apicast.policy')
+local _M = policy.new('{{ policy.summary }}')
```
Comment (davidor) [inline examples/scaffold/policy/gateway/src/apicast/policy/{{policy.file}}/{{policy.file}}.lua]: Shouldn't this be `policy.name` ?

File: gateway/Roverfile.lock [hunk 2]
```diff
@@ -1,4 +1,4 @@
-argparse 0.5.0-1||production
+argparse 0.6.0-1||production
```
Comment (davidor) [inline gateway/Roverfile.lock]: Do we need the new version for some reason?

File: examples/scaffold/policy/t/apicast-policy-{{policy.file}}.t [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+=== TEST 1: {{ policy.name}} accepts configuration
+--- configuration
+{
+  "services": [
+    {
+      "proxy": {
+        "policy_chain": [
+          { "name": "apicast.policy.{{ policy.file }}",
+            "configuration": { } },
+          { "name": "apicast.policy.echo" }
```
Comment (davidor) [inline examples/scaffold/policy/t/apicast-policy-{{policy.file}}.t]: Why is this needed?

File: gateway/src/apicast/cli/command/generate/policy.lua [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      name = options.name,
+      version = options.version,
+      summary = options.summary,
+    }
+  }, options.force)
+end
+
+local directory = function(dir) if pl.path.dir(dir) then return pl.path.abspath(dir) end end
+
+return function(parser)
```
Comment (davidor) [inline gateway/src/apicast/cli/command/generate/policy.lua]: I think the option for `summary` is missing.

File: gateway/src/apicast/cli/command/generate.lua [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+function _M.new(parser)
+  local cmd = configure(parser:command('generate', 'Generate scaffolding'))
+
+  return setmetatable({ parser = parser, cmd = cmd }, mt)
+end
+
+function _M.copy(source, destination, env, force)
+  print('source: ', source)
+  print('destinatioon: ', destination)
```
Comment (davidor) [inline gateway/src/apicast/cli/command/generate.lua]: typo

**Comment by davidor** [comment]
Looks good :+1: 
We needed something like this.

Can you mention this option somewhere in the `policies.md` doc?

File: gateway/Roverfile.lock (hunk 2 above)
Comment (mikz) [reply gateway/Roverfile.lock]: No good reason. rover is not so smart about releasing locked versions.

File: examples/scaffold/policy/gateway/src/apicast/policy/{{policy.file}}/{{policy.file}}.lua (hunk 1 above)
Comment (mikz) [reply examples/scaffold/policy/gateway/src/apicast/policy/{{policy.file}}/{{policy.file}}.lua]: Yep. It should be some human name.
